core/src/main/java/io/neow3j/protocol/rx/JsonRpc2_0Rx.java

```diff
@@ -69,73 +84,73 @@ private Observable<NeoGetBlock> replayBlocksObservableSync(BigInteger startBlock
             return Observable.error(e);
         }
 
-        if (startBlockNumber.compareTo(latestBlockNumber) > -1) {
-            return onCompleteObservable;
+        if (startBlockIdx.compareTo(latestBlockIdx) > -1) {
+            return onCaughtUpObservable;
         } else {
             return Observable.concat(
-                    replayBlocksObservableSync(startBlockNumber, latestBlockNumber, fullTransactionObjects),
+                    replayBlocksObservableSync(startBlockIdx, latestBlockIdx, fullTransactionObjects, true),
                     Observable.defer(() -> catchUpToLatestBlockObservableSync(
-                            latestBlockNumber.add(BigInteger.ONE),
+                            latestBlockIdx.add(BigInteger.ONE),
                             fullTransactionObjects,
-                            onCompleteObservable)));
+                            onCaughtUpObservable)));
         }
     }
 
     public Observable<Transaction> catchUpToLatestTransactionObservable(BigInteger startBlock) {
         return catchUpToLatestBlockObservable(startBlock, true, Observable.empty())
-                .flatMapIterable(JsonRpc2_0Rx::toTransactions);
+                .flatMapIterable(b -> b.getBlock().getTransactions());
     }
 
+    /**
+     * Creates an observable that emits blocks starting at {@code startBlockNumber} up to the most recent block and
+     * continues emitting blocks that are newly created on the Neo blockchain. The new blocks are pulled every
+     * {@code pollingInterval}.
+     *
+     * Do not use {@code retry(...)} on this observable because it will not only resubscribe to the new blocks but
+     * also replay the blocks since {@code startBlock}.
+     *
+     * @param startBlock             The block index at which to start catching up.
+     * @param fullTransactionObjects If the full transactions objects should be included in the blocks.
+     * @param pollingInterval        The polling interval in milliseconds.
+     * @return the block observable.
+     */
     public Observable<NeoGetBlock> catchUpToLatestAndSubscribeToNewBlocksObservable(BigInteger startBlock,
             boolean fullTransactionObjects, long pollingInterval) {
 
         return catchUpToLatestBlockObservable(startBlock, fullTransactionObjects,
                 blockObservable(fullTransactionObjects, pollingInterval));
     }
 
-    public Observable<NeoGetBlock> blockObservable(boolean fullTransactionObjects, long pollingInterval) {
-        return neoBlockObservable(pollingInterval)
-                .flatMap(blockIndex ->
-                        neow3j.getBlock(blockIndex, fullTransactionObjects).observable());
-    }
-
-    public Observable<NeoGetBlock> subscribeToNewBlocksObservable(boolean fullTransactionObjects, long pollingInterval)
-            throws IOException {
-
-        return catchUpToLatestAndSubscribeToNewBlocksObservable(getLatestBlockNumber(), fullTransactionObjects,
-                pollingInterval);
-
-    }
```

Comment:
This can be replaced by `blockObservable`. It starts emitting blocks at the latest block. Calling `catchUpToLatestAndSubscribeToNewBlocksObservable` with the latest block index as param results in the same as calling `blockObservable`.